packages/web3-providers-ws/src/types.ts

```diff
@@ -0,0 +1,29 @@
+import { OutgoingHttpHeaders } from 'http';
+import { JsonRpcPayload, JsonRpcResponse } from 'web3-common';
+import { IClientConfig } from 'websocket';
+import { DeferredPromise } from './deferredPromise';
+
+export type ReconnectOptions = {
+	auto: boolean;
+	delay: number;
+	maxAttempts: number;
+	onTimeout: boolean;
+};
+
+export interface WebSocketOptions {
+	protocol?: string;
+	headers?: OutgoingHttpHeaders;
+	/* eslint-disable @typescript-eslint/ban-types */
+	requestOptions?: Object;
```

Comment:
```suggestion
	requestOptions?: Record<string, unknown>;
```

[Typescript docs](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html) say to not use `Object` as a type. Additionally, we should know what the options are and make this strongly typed (instead of using `Record<string, unknown>`)

Reply:
with new lib isomorphic-ws and ws its no longer required.